src/check-c-linkage-decls.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+
+import sys, os
+
+os.chdir (os.environ.get ('srcdir', os.path.dirname (__file__)))
+
+HBHEADERS = os.environ.get ('HBHEADERS')
+if HBHEADERS:
+	HBHEADERS = HBHEADERS.split ()
+else:
+	HBHEADERS = [x for x in os.listdir ('.') if x.startswith ('hb') and x.endswith ('.h')]
```

Comment:
I think I like Khaled's other suggestion to rewrite these as:
```
headers = os.environ.get ('HBHEADERS', '').split ()
if not headers:
        headers = ...
```

These repetitive 5-line blocks are annoying.

Reply:
Reduced to a one liner using `or` :)

Reply:
Hehe.  BTW, you can also omit the `'.'` to `os.listdir`.

Actually the original scripts pass a `maxdepth=1` to `find`.  So you want `glob` here, which will further simplify things eg. `glob.glob ('hb*.h')`... 

The original script was using `find` for some other reasons.  I see you are `chdir`ing into `srcdir`.  All of this had good reasons for how they were done.  Can you please test this with out-of-tree builds with autotools and make sure it works?